### ISSUE TITLE
Cycled iterable constructed from iterator cycles only once

### ISSUE THREAD

**Issue by neonailol** [yegor256/cactoos#905]
I have a problem with iterable/Cycled that is constructed from iterator, 
https://github.com/yegor256/cactoos/blob/c6ca6d183ee5d0ca7dd76701231333af611a6439/src/main/java/org/cactoos/iterable/Cycled.java#L54
it does not cycle when constructed using this constructor
I tested it with the following code:

```java
@Test
    public void repeatIteratorTest() throws Exception {
        final String expected = "two";
        MatcherAssert.assertThat(
            "Can't repeat iterator",
            new ItemAt<>(
                // @checkstyle MagicNumberCheck (1 line)<
                7, new Cycled<>(
                new IteratorOf<>(
                    "one", expected, "three"
                )
            )
            ),
            new ScalarHasValue<>(
                expected
            )
        );
    }
```

Am i missing something?

**Comment by 0crat** [comment]
@llorllale[/z](https://www.0crat.com/u/llorllale) please, pay attention to this issue

**Comment by 0crat** [comment]
@neonailol[/z](https://www.0crat.com/u/neonailol) this project will fix the problem faster if you donate a few dollars to it; just [click here](https://www.0crat.com/contrib/C63314D6Z) and pay via Stripe, it's very fast, convenient and appreciated; thanks a lot!

**Comment by llorllale** [comment]
@0crat in

**Comment by 0crat** [comment]
> @0crat in ([here](https://github.com/yegor256/cactoos/issues/905#issuecomment-395202965))

@llorllale Job [#905](https://github.com/yegor256/cactoos/issues/905) is now in scope, role is `DEV`

**Comment by 0crat** [comment]
Bug was reported, see [§29](http://www.zerocracy.com/policy.html#29): +15 point(s) just awarded to @neonailol[/z](https://www.0crat.com/u/neonailol)

**Comment by 0crat** [comment]
@llorllale[/z](https://www.0crat.com/u/llorllale) not enough funds available in the project, can't set budget of job [#905](https://github.com/yegor256/cactoos/issues/905), see [§21](http://www.zerocracy.com/policy.html#21); @baybatu[/z](https://www.0crat.com/u/baybatu) will get no money on completion; in order to fix that, add funds to the project and assign the job again

**Comment by 0crat** [comment]
The user @baybatu[/z](https://www.0crat.com/u/baybatu) resigned from [#905](https://github.com/yegor256/cactoos/issues/905), please stop working. Reason for job resignation: It is older than 10 days, see [§8](http://www.zerocracy.com/policy.html#8)

**Comment by 0crat** [comment]
Resigned on delay, see [§8](http://www.zerocracy.com/policy.html#8): -30 point(s) just awarded to @baybatu[/z](https://www.0crat.com/u/baybatu)

**Comment by g4s8** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/yegor256/cactoos/issues/905#issuecomment-408181623))

@g4s8 This is what I know about this job in [C63314D6Z](https://www.0crat.com/p/C63314D6Z), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#905](https://github.com/yegor256/cactoos/issues/905) is [in scope](http://www.zerocracy.com/policy.html#14) for 2mon
  * The role is `DEV`
  * The job is not assigned to anyone
  * The [budget](http://www.zerocracy.com/policy.html#4) is not set yet
  * These users are banned and won't be assigned:
    * @neonailol[/z](https://www.0crat.com/u/neonailol): This user reported the ticket
    * @baybatu[/z](https://www.0crat.com/u/baybatu): User was resigned from the ticket
  * Job [footprint](https://www.0crat.com/footprint/C63314D6Z?q=%7Bjob%3A%27gh%3Ayegor256%2Fcactoos%23905%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by 0crat** [comment]
The job [#905](https://github.com/yegor256/cactoos/issues/905) assigned to @tomascury[/z](https://www.0crat.com/u/tomascury), here is [why](https://www.0crat.com/footprint/C63314D6Z/67182b96-3f2b-4438-a816-feb20d119cce); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be no monetary reward for this job

**Comment by tomascury** [comment]
@0crat resign

**Comment by tomascury** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/yegor256/cactoos/issues/905#issuecomment-464515652))

@tomascury The user @tomascury[/z](https://www.0crat.com/u/tomascury) resigned from [#905](https://github.com/yegor256/cactoos/issues/905), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
> @0crat resign ([here](https://github.com/yegor256/cactoos/issues/905#issuecomment-464434719))

@tomascury Job `gh:yegor256/cactoos#905` is not assigned, can't get performer

**Comment by 0crat** [comment]
Tasks refusal is discouraged, see [§6](http://www.zerocracy.com/policy.html#6): -15 point(s) just awarded to @tomascury[/z](https://www.0crat.com/u/tomascury)

**Comment by 0crat** [comment]
The job [#905](https://github.com/yegor256/cactoos/issues/905) assigned to @borysfan[/z](https://www.0crat.com/u/borysfan), here is [why](https://www.0crat.com/footprint/C63314D6Z/31c0f15c-e8d2-44b5-8b16-8cfe039c63d8); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be no monetary reward for this job

**Comment by llorllale** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/yegor256/cactoos/issues/905#issuecomment-474141757))

@llorllale The user @borysfan[/z](https://www.0crat.com/u/borysfan) resigned from [#905](https://github.com/yegor256/cactoos/issues/905), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
The job [#905](https://github.com/yegor256/cactoos/issues/905) assigned to @victornoel[/z](https://www.0crat.com/u/victornoel), here is [why](https://www.0crat.com/footprint/C63314D6Z/709029d6-41dd-4974-902f-09b834dd4e4a); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be no monetary reward for this job

**Comment by victornoel** [comment]
@llorllale what is the status of this since #911 was accepted? This constructor should be removed in #1059...

**Comment by llorllale** [comment]
@0crat out

**Comment by 0crat** [comment]
> @0crat out ([here](https://github.com/yegor256/cactoos/issues/905#issuecomment-476428751))

@llorllale GitHub issue is still open, won't close

**Comment by llorllale** [comment]
@0crat refuse

**Comment by llorllale** [comment]
@0crat out

**Comment by 0crat** [comment]
> @0crat out ([here](https://github.com/yegor256/cactoos/issues/905#issuecomment-476431148))

@llorllale Job `gh:yegor256/cactoos#905` is not assigned, can't get performer

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/yegor256/cactoos/issues/905#issuecomment-476431137))

@llorllale The user @victornoel[/z](https://www.0crat.com/u/victornoel) resigned from [#905](https://github.com/yegor256/cactoos/issues/905), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
> @0crat out ([here](https://github.com/yegor256/cactoos/issues/905#issuecomment-476431148))

@llorllale GitHub issue is still open, won't close

**Comment by 0crat** [comment]
The job [#905](https://github.com/yegor256/cactoos/issues/905) assigned to @atapin[/z](https://www.0crat.com/u/atapin), here is [why](https://www.0crat.com/footprint/C63314D6Z/87f49133-fe2e-4951-b035-9079dba339bb); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be no monetary reward for this job

**Comment by atapin** [comment]
@llorllale it looks like the scope of #911 makes this issue outdated

**Comment by atapin** [comment]
@llorllale ping

**Comment by llorllale** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/yegor256/cactoos/issues/905#issuecomment-480094191))

@llorllale The user @atapin[/z](https://www.0crat.com/u/atapin) resigned from [#905](https://github.com/yegor256/cactoos/issues/905), please stop working. Reason for job resignation: Order was cancelled

**Comment by llorllale** [comment]
@neonailol this ticket is no longer current - see #911 for reason as to why

**Comment by llorllale** [comment]
@0crat out

**Comment by 0crat** [comment]
> @0crat out ([here](https://github.com/yegor256/cactoos/issues/905#issuecomment-480096025))

@llorllale Job `gh:yegor256/cactoos#905` is not assigned, can't get performer

**Comment by 0crat** [comment]
> @0crat out ([here](https://github.com/yegor256/cactoos/issues/905#issuecomment-480096025))

@llorllale The job [#905](https://github.com/yegor256/cactoos/issues/905) is now out of scope

**Comment by 0crat** [comment]
Job `gh:yegor256/cactoos#905` is not assigned, can't get performer

**Comment by 0crat** [comment]
This job is not in scope
